programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1147,7 +1148,11 @@ static void on_event(const base_event *ev) {
                         for (int i = 0; svc->Addresses[i]; i++) {
                             tunnel_address *addr = svc->Addresses[i];
                             if (addr->IsHost && model_map_get(hostnamesToAdd, addr->HostName) == NULL) {
-                                model_map_set(hostnamesToAdd, addr->HostName, true);
+                                if (model_map_get(hostnamesToRemove, addr->HostName) != NULL) {
+                                    model_map_set(hostnamesToEdit, addr->HostName, true);
+                                } else {
+                                    model_map_set(hostnamesToAdd, addr->HostName, true);
```

Comment:
<img width="568" alt="image" src="https://user-images.githubusercontent.com/13525299/153798308-8e91998f-0257-44b2-99bb-c9894213034c.png">

It's ok to use the map as a list (of keys), but you can set the values to NULL instead of using an incompatible type for the value.

Reply:
I want to check if the value is already added to the map, so I am verifying the map value != NULL in the list 1151 - model_map_get(hostnamesToRemove, addr->HostName) != NULL). So I will use a static string instead of true